src/Newton.jl

```diff
@@ -113,7 +113,9 @@ function newton(Fhandle, Jhandle, x0, options:: NewtonPar{T}; normN = norm) wher
 	@unpack tol, maxIter, verbose, linesearch = options
 
 	# Initialise iterations
-	x = convert(typeof(Fhandle(x0)),copy(x0))
+	x = zeros(T,length(x0))
+	x .= copy(x0)
+
```

Comment:
@rveltz is this ok? I need x to be of type `Array{TrackedReal}` not `TrackedArray`. This means that the array is mutable and there fore compatible with all downstream in-place operations.

Reply:
Do you get an error if not? 

Why dont you call `newton(F, convert(T, x0),...)`? All this wrapping is specific to your implementation. You could also make this wrapping in `F`...

Reply:
yes we get a `setIndex!` error in `minus!`

Reply:
`convert(T, x0)` still gives you a `TrackedArray` which is not desirable

Reply:
wait nevermind.. i could set `T=Array{TrackedReal}`..

Reply:
There are several things:
- you can probably call `newton(F, convert(Array{TrackedReal}, x0),...)` or make the proper conversion before calling `newton`. 
- you can overload `minus!` as I showed in [here](https://github.com/gszep/FluxContinuation/tree/minimal-working-example)
- **However**, all this does not cure the origin of the issue which is: when you use finite differences with your `F` to get a jacobian `J`, the type of `J \ f` is not the same as the type of `f`.

Reply:
yep this works! thank you for you patience.. I'm so crap at types :man_facepalming: